CRM/Mailing/BAO/Mailing.php

```diff
@@ -2403,26 +2403,24 @@ public static function showEmailDetails($id) {
    *   Id of the mail to delete.
    *
    * @return void
+   *
+   * @deprecated
    */
   public static function del($id) {
-    if (empty($id)) {
-      throw new CRM_Core_Exception(ts('No id passed to mailing del function'));
-    }
-
-    CRM_Utils_Hook::pre('delete', 'Mailing', $id);
-
-    // delete all file attachments
-    CRM_Core_BAO_File::deleteEntityFile('civicrm_mailing',
-      $id
-    );
-
-    $dao = new CRM_Mailing_DAO_Mailing();
-    $dao->id = $id;
-    $dao->delete();
-
+    static::deleteRecord(['id' => $id]);
     CRM_Core_Session::setStatus(ts('Selected mailing has been deleted.'), ts('Deleted'), 'success');
```

Comment:
What?? Form-layer function `setStatus()` called from a business-layer function? Oh the horror!

We need to delete that line and add it to where the function is called, on the next line after here:

https://github.com/civicrm/civicrm-core/blob/7d0abbd1f48255943f3bed9eae27ec3249b9e55b/CRM/Mailing/Page/Browse.php#L202

 and here:

https://github.com/civicrm/civicrm-core/blob/bc77d7c0798081f999e6eca92d75fa874fd04e74/CRM/Mailing/Form/Browse.php#L64

Reply:
(I found another form that calls the function too but it sets its own status message so the user would see double, just deleting it here will fix that)